src/dnmd/internal.h

```diff
@@ -117,6 +117,13 @@ typedef struct _mdtable_t
 
 typedef mdcdata_t mdstream_t;
 
+typedef struct _mdmem_t
+{
+    struct _mdmem_t* next;
+    size_t size;
+    uint8_t data[1]; // Arbitrary sized array
```

Comment:
Use a zero-length array to make it clearer that this goes off the end of the struct?

```suggestion
    uint8_t data[0]; // Arbitrary sized array
```

Reply:
Sadly that is non-standard in both C and C++. Arrays must be declared with a non-zero value. There is the C99 `*` declaration but I think that is limited to signatures. I don't think I've ever observed that in a field declaration.